zebra-chain/src/block/header.rs

```diff
@@ -81,11 +81,36 @@ impl BlockHeader {
 
         Ok(())
     }
+
+    /// Check if `self.time` is less than or equal to
+    /// 2 hours in the future, according to the node's local clock (`now`).
+    ///
+    /// This is a non-deterministic rule, as clocks vary over time, and
+    /// between different nodes.
+    ///
+    /// "In addition, a full validator MUST NOT accept blocks with nTime
+    /// more than two hours in the future according to its clock. This
+    /// is not strictly a consensus rule because it is nondeterministic,
+    /// and clock time varies between nodes. Also note that a block that
+    /// is rejected by this rule at a given point in time may later be
+    /// accepted."[S 7.5][7.5]
+    ///
+    /// [7.5]: https://zips.z.cash/protocol/protocol.pdf#blockheader
+    pub fn is_time_valid_local_clock(&self, now: DateTime<Utc>) -> Result<(), Error> {
```

Comment:
TIOLI: the `local_clock` part of this name confused me on first reading because I thought that the `local` was a reference to local time (as in time zone).  WDYT of `is_time_valid_at`?

Reply:
Fixed in e3a823b4e108d1871bb0fedd4b7d65a1fa2c0b19